src/CodeGen_Hexagon.cpp

```diff
@@ -371,9 +370,6 @@ llvm::Function *CodeGen_Hexagon::define_hvx_intrinsic(Intrinsic::ID id, Type ret
     llvm::Function *intrin = Intrinsic::getDeclaration(module.get(), id);
     return define_hvx_intrinsic(intrin, ret_ty, name, arg_types, broadcast_scalar);
 }
-llvm::Value *CodeGen_Hexagon::shl_or(llvm::Value *v, unsigned u) {
-    return builder->CreateOr(builder->CreateShl(v, u), v);
- }
 llvm::Function *CodeGen_Hexagon::define_hvx_intrinsic(llvm::Function *intrin, Type ret_ty, const std::string &name,
                                                       const std::vector<Type> &arg_types, bool broadcast_scalar) {
```

Comment:
Sorry, one more thing: can you rename broadcast_scalar to broadcast_scalar_to_word? Or something like that. It's ambiguous whether that means broadcast the scalar all the way up to a vector, or just to a word like it's doing.